### ISSUE TITLE
Build Update/Upload Issues Over Wireless

### ISSUE THREAD

**Issue by nmiller0113** [dpitts/f16v4_issues#14]
I have an F48v4 on build 12. Every time I try to upgrade the firmware to build 13 I get a firmware upload failed message and then the F48 becomes unavailable until I reboot, but I've also got it to take the firmware upload and then proceed to say firmware invalid press a key and then it comes back on build 12. I've tried this three times on two different browsers and even tried downloading the firmware a second time to see if it was a corrupted download. I've had no issues upgrading prior to this, and the controller operates fine after the reboot. I also should note that I'm performing these upgrades over wifi, and I don't believe I had done it that way previously. The wifi has been stable and fpp updates have gone fine and it is in the same box, so I'm not sure that is related.

It seems reproducible if someone would like to test with me. Thank you!

**Comment by keithsw1111** [comment]
Can you apply the update using an SD card/wired connection ... some of the fixes in 13 are actually wifi related.

**Comment by nmiller0113** [comment]
I applied the update without issue using the SD Card method. I'll try it again over wireless the next time a new version comes out. Thanks!

**Comment by nmiller0113** [comment]
@keithsw1111 I just tried to update to Build 14 over wireless and it still fails saying "Firmware Invalid" on the LED screen of my F48v4. Unfortunately, the wireless fixes you did in Build 13 did not correct it. I'd really really like to be able to upgrade firmware over wireless as the controller is stored in a spot that isn't easy to access and this limitation will make things tricky. I'll navigate it if needed :) but hopefully, we can find a fix. I'm assuming the problem is the file getting corrupted during transfer. My Wireless signal is super strong and I've got no problems managing it otherwise.

**Comment by keithsw1111** [comment]
There is definitely some inconsistency here.  It is working sometimes for me but others i do see a failure.

**Comment by nmiller0113** [comment]
That's interested @keithsw1111. I hope you're able to narrow down where the issue is.

**Comment by keithsw1111** [comment]
I may have a fix. Ping me on email if you want to try it.

**Comment by keithsw1111** [comment]
After discussing this with nathan Network topology seems to be a factor. The scenario we are looking into is:

Computer --(wired)->Switch --(wired)-> AP --(wifi)-> controller firmware upload fails.

**Comment by nmiller0113** [comment]
@keithsw1111, my apologies for not getting back to the debugging on this sooner. Been a busy half year so far! I noticed several build release notes talking about issues resolved similar to what I was experiencing on build 14, so I decided to upgrade to the latest, build 20. After upgrading to build 20, it appears the upgrade over wifi issues have been resolved. I'm closing this out! Thank you for your hard work on this @keithsw1111!
